pkg/worker/lifecycle.go

```diff
@@ -511,3 +520,47 @@ func (s *Worker) createOverlay(request *types.ContainerRequest, bundlePath strin
 func (s *Worker) isBuildRequest(request *types.ContainerRequest) bool {
 	return request.SourceImage != nil
 }
+
+func (s *Worker) watchOOMEvents(ctx context.Context, containerId string, output chan common.OutputMsg) {
+	time.Sleep(time.Second)
```

Comment:
This feels like it could be a race condition, is there a way we could make this event based / only start watching when we're sure the container is up?

Reply:
We wait for the pid in the spawn function. For some reason, if we don't sleep or retry making the events channel, the channel can be closed right after we just made it. 

I updated the logic to where we don't have random sleeps and can retry making the events channel up to 5 times before bailing.